sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -518,3 +518,13 @@ def test_agg_n_clusters():
         msg = ("n_clusters should be an integer greater than 0."
                " %s was provided." % str(agc.n_clusters))
         assert_raise_message(ValueError, msg, agc.fit, X)
+
+
+def test_affinity_passed_to_fix_connectivity():
+    # Test that the affinity parameter is actually passed to the pairwise
+    # function
+
+    rng = np.random.RandomState(42)
+    X = rng.normal(size=(5, 5))
+
+    assert_raises(ValueError, linkage_tree, X, affinity='FakeAffinity')
```

Comment:
I'm pretty sure this test will pass at master, i.e. an error will be raised by another part of `linkage_tree`.